zephyr/include/rtos/alloc.h

```diff
@@ -0,0 +1,143 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __ZEPHYR_RTOS_ALLOC_H__
+#define __ZEPHYR_RTOS_ALLOC_H__
+
+#include <rtos/bit.h>
+#include <rtos/string.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+/** \addtogroup alloc_api Memory Allocation API
```

Comment:
This doesn't seem right. rmalloc() and friends is really a SOF specific "audio customer allocator" built on top of the RTOS allocator. I don't think this should be under rtos moniker, but rather a SOF service. 

We can discuss whether app code should use Zephyr allocator directly, but currently that is not even possible as you need some intermedia to set up the heaps in the way SOF expects to (currently done in wrapper.c).

Reply:
This is going towards a more POSIX like API, but it need some de-tangling before that can happen.